flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/window/slicing/SlicingWindowOperator.java

```diff
@@ -150,6 +158,8 @@ public void open() throws Exception {
                         getKeyedStateBackend(),
                         collector,
                         getRuntimeContext()));
+        // initialize progress of window processor
+        windowProcessor.advanceProgress(currentWatermark);
```

Comment:
I think we should not call advanceProgress here, because it may be possible that advanceProgress would generate outputs, which is not allowed in open. maybe just add an initialProgess paramter in the constructor?

Reply:
Currently, there is no windowProcessor implementation flushes data to downstream.